packages/zoe/src/contractFacet/zcfSeat.js

```diff
@@ -282,59 +300,75 @@ export const createSeatManager = (
         const assetKind = getAssetKindByBrand(brand);
         return AmountMath.makeEmpty(brand, assetKind);
       },
-      getCurrentAllocation: () => getCurrentAllocation(zcfSeat),
-      getStagedAllocation: () => getStagedAllocation(zcfSeat),
-      isOfferSafe: newAllocation => {
-        assertActive(zcfSeat);
-        const currentAllocation = getCurrentAllocation(zcfSeat);
+      getCurrentAllocation: ({ self }) => getCurrentAllocation(self),
+      getStagedAllocation: ({ self }) => getStagedAllocation(self),
+      isOfferSafe: ({ state, self }, newAllocation) => {
+        assertActive(self);
+        const currentAllocation = getCurrentAllocation(self);
         const reallocation = harden({
           ...currentAllocation,
           ...newAllocation,
         });
 
-        return isOfferSafe(proposal, reallocation);
+        return isOfferSafe(state.proposal, reallocation);
       },
-      incrementBy: amountKeywordRecord => {
-        assertActive(zcfSeat);
+      incrementBy: ({ self }, amountKeywordRecord) => {
+        assertActive(self);
         amountKeywordRecord = coerceAmountKeywordRecord(
           amountKeywordRecord,
           getAssetKindByBrand,
         );
         setStagedAllocation(
-          zcfSeat,
-          addToAllocation(getStagedAllocation(zcfSeat), amountKeywordRecord),
+          self,
+          addToAllocation(getStagedAllocation(self), amountKeywordRecord),
         );
         return amountKeywordRecord;
       },
-      decrementBy: amountKeywordRecord => {
-        assertActive(zcfSeat);
+      decrementBy: ({ self }, amountKeywordRecord) => {
+        assertActive(self);
         amountKeywordRecord = coerceAmountKeywordRecord(
           amountKeywordRecord,
           getAssetKindByBrand,
         );
         setStagedAllocation(
-          zcfSeat,
+          self,
           subtractFromAllocation(
-            getStagedAllocation(zcfSeat),
+            getStagedAllocation(self),
             amountKeywordRecord,
           ),
         );
         return amountKeywordRecord;
       },
-      clear,
-      hasStagedAllocation: () => hasStagedAllocation(zcfSeat),
-    });
+      clear: dropContext(clear),
+      hasStagedAllocation: ({ self }) => hasStagedAllocation(self),
+    },
+  );
+  const makeZCFSeat = ({
+    proposal,
+    notifier: _notifier,
```

Comment:
is `notifier` here to document the shape of the first parameter? please use jsdoc instead